app/controllers/emails_controller.rb

```diff
@@ -18,23 +23,96 @@ def create
     # rails-method-that-converts-data-from-datetime-select-into-a-datet
 
     # extract send datetime
-    send_date = Date.new(email["created_at(1i)"].to_i,
-                        email["created_at(2i)"].to_i,
-                        email["created_at(3i)"].to_i)
+    sendDate = Date.new(email["sendDate(1i)"].to_i,
+                        email["sendDate(2i)"].to_i,
+                        email["sendDate(3i)"].to_i)
     # extract subject of the email
     subject = email["subject"]
 
     # extract content of the email
     content = email["description"]
+    @email = Email.new(
+      subject: subject, 
+      description: content, 
+      sendDate: sendDate,
+      sent_status: false,
+      send_to_never_logged_in: email["send_to_never_logged_in"],
+      send_to_never_pulled_room: email["send_to_never_pulled_room"],
+      send_to_formerly_in_room: email["send_to_formerly_in_room"],
+      send_to_in_room: email["send_to_in_room"],
+      send_to_admins: email["send_to_admins"])
 
-    @email = Email.new(subject: subject, description: content, sendDate: send_date)
-    @email.save
+    respond_to do |format|
+      if @email.save
+        # Run bin/delayed_job start to process all jobs
+        GeneralMailer.reminder_email_to_non_participants(sendDate, @email.attributes['id'])
+        format.html { redirect_to emails_show_path, notice: 'Email was successfully created.' }
+        format.json { render :show, status: :ok}
+      else
+        format.html { render :new }
+        format.json { render json: @email.errors, status: :unprocessable_entity }
+      end
+    end
+  end
 
-    # Run bin/delayed_job start to process all jobs
-    GeneralMailer.reminder_email_to_non_participants(subject, content, send_date)
+  def update
+    respond_to do |format|
+      email = params["email"]
+      params["email"]["sendDate"] = Date.new(email["sendDate(1i)"].to_i,
```

Comment:
If the sendDate of an email changes, do we need to set a new delayed_job to process it?

Reply:
Fixed!